test/unit/utils/test-xhr-utils.js

```diff
@@ -0,0 +1,204 @@
+    });
+  });
+
+  describe('setupAMPCors', () => {
+    describe('requireAmpResponseSourceOrigin', () => {
+      it('should be false if ampCors is false', () => {
+        const fetchInitDef = setupAMPCors(
+            {origin: 'http://www.origin.org'}, 'http://www.origin.org', {ampCors: false});
+        expect(fetchInitDef.requireAmpResponseSourceOrigin).to.equal(false);
+      });
+
+      it('should be set if not defined', () => {
+        const fetchInitDef = setupAMPCors(
+            {origin: 'http://www.origin.org'}, 'http://www.origin.org', {});
+        expect(fetchInitDef.requireAmpResponseSourceOrigin).to.equal(true);
+      });
+    });
+
+    it('should set AMP-Same-Origin header', () => {
+      // Given a same origin request.
+      const fetchInitDef = setupAMPCors(
+          {origin: 'http://www.origin.org'}, 'http://www.origin.org', {});
+      // Expect proper header to be set.
+      expect(fetchInitDef['headers']['AMP-Same-Origin']).to.equal('true');
+    });
+
+    it('should not set AMP-Same-Origin header', () => {
+      // If not a same origin request.
+      const fetchInitDef = setupAMPCors(
+          {origin: 'http://www.originz.org'}, 'http://www.origin.org', {headers: {}});
+      expect(fetchInitDef['headers']['AMP-Same-Origin']).to.be.undefined;
+    });
+  });
+
+  describe('setupInit', () => {
+    it('should set up init', () => {
+      const init = setupInit();
+      expect(init).to.deep.equal({method: 'GET', headers: {}});
+    });
+
+    it('should set up init with Accept header value', () => {
+      const init = setupInit(undefined, 'text/html');
+      expect(init['headers']['Accept']).to.equal('text/html');
+    });
+
+    it('should handle null credentials', () => {
+      allowConsoleError(() => { expect(() => {
+        setupInit({credentials: null}, 'text/html');
+      }).to.throw(/Only credentials=include\|omit support: null/); });
+    });
+  });
+
+  describe('setupJsonFetchInit', () => {
+    it('set proper properties', () => {
+      expect(setupJsonFetchInit({body: {}})).to.deep.equal({
+        headers: {
+          'Accept': 'application/json',
+        },
+        body: {},
+        method: 'GET',
+      });
+
+      expect(setupJsonFetchInit({body: {}, method: 'POST'})).to.deep.equal({
+        headers: {
+          'Accept': 'application/json',
+          'Content-Type': 'text/plain;charset=utf-8',
+        },
+        body: '{}',
+        method: 'POST',
+      });
+    });
+  });
+
+  describe('getViewerInterceptResponse', () => {
+    let viewer;
+    let viewerForDoc;
+    beforeEach(() => {
+      viewer = {
+        hasCapability: unusedParam => false,
+        whenFirstVisible: () => Promise.resolve(),
+        sendMessageAwaitResponse: sandbox.stub(),
+      };
+      viewerForDoc = sandbox.stub(Services, 'viewerForDoc');
+    });
+
+    it('should be no-op if amp doc is absent', () => {
+      return getViewerInterceptResponse({}, null, '', {}).then(() => {
```

Comment:
Add inline comments or use local vars to describe what we're passing in.

```js
getViewerInterceptResponse(/* win */ {}, /* ampdocSingle */, null, /* input */ '', /* init */ {})
```

Or:
```js
const win = {},
const ampdocSingle = null;
const input = '';
const init = {};
getViewerInterceptResponse(win, ampdocSingle, input, init);
```

You could also define these in `beforeEach` to so each `it` test can use them.